src/WebJobs.Script/Description/CSharp/FunctionMetadataResolver.cs

```diff
@@ -37,9 +37,15 @@ public sealed class FunctionMetadataResolver : MetadataReferenceResolver
                 "System.Xml",
                 "System.Net.Http",
                 typeof(object).Assembly.Location,
-                typeof(TraceWriter).Assembly.Location,
-                typeof(TimerInfo).Assembly.Location,
-                typeof(System.Web.Http.ApiController).Assembly.Location
+                typeof(IAsyncCollector<>).Assembly.Location, /*Microsoft.Azure.WebJobs*/
+                typeof(JobHost).Assembly.Location, /*Microsoft.Azure.WebJobs.Host*/
+                typeof(CoreJobHostConfigurationExtensions).Assembly.Location, /*Microsoft.Azure.WebJobs.Extensions*/
+                typeof(System.Web.Http.ApiController).Assembly.Location /*System.Web.Http*/
+            };
+
+        private static readonly Assembly[] SharedAssemblies =
+            {
+                typeof(Newtonsoft.Json.JsonConvert).Assembly /*Newtonsoft.Json*/
             };
```

Comment:
Oh, I see you explain that in the PR! Maybe add one line comment on SharedAssemblies, e.g. "Assemblies that are not referenced by default but can be explicitly referenced using `#r`"